### PR TITLE
Search results design tweaks

### PR DESCRIPTION
**Pull request by vovakulikov**
Fixes https://github.com/sourcegraph/sourcegraph/issues/59939

## TODO 
- [x] Adjust zero state for the inline filters search 
- [x] Adjust "more results" description look (but we still need to update text there cc @taiyab)
- [x] Move action buttons to bottom 
- [x] Add gradient background and top border for the hover footer state, cc @taiyab can you please provide a gradient value for these two (background and border) I tried to use our standard `--marketing-gradient` and it looked off 
- [x] Add placeholder UI (for everything but utility section since utility much faster and should not affect placeholder appearance) cc @taiyab maybe you also have something about this placeholder already. If not I will use [something like this](https://codesandbox.io/p/sandbox/skeleton-placeholder-0y8od?file=%2Findex.html&from-embed=) gradient with a few placeholders lines 

## Test plan
- Check that zero states in inline filter search looks good
- Check that skeleton shows up and looks properly 
- Check footer doc hover state

### CODE REVIEW

**Comment by taiyab** [comment]
@vovakulikov We can introduce a gradient style here, and I'll go back with @rrhyne and unify them all.

For the loader:
https://www.svgbackgrounds.com/elements/animated-svg-preloaders/

`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 200 200"><rect fill="#0B70DB" stroke="#0B70DB" stroke-width="16" width="30" height="30" x="25" y="85"><animate attributeName="opacity" calcMode="spline" dur="2" values="1;0;1;" keySplines=".5 0 .5 1;.5 0 .5 1" repeatCount="indefinite" begin="-.4"></animate></rect><rect fill="#0B70DB" stroke="#0B70DB" stroke-width="16" width="30" height="30" x="85" y="85"><animate attributeName="opacity" calcMode="spline" dur="2" values="1;0;1;" keySplines=".5 0 .5 1;.5 0 .5 1" repeatCount="indefinite" begin="-.2"></animate></rect><rect fill="#0B70DB" stroke="#0B70DB" stroke-width="16" width="30" height="30" x="145" y="85"><animate attributeName="opacity" calcMode="spline" dur="2" values="1;0;1;" keySplines=".5 0 .5 1;.5 0 .5 1" repeatCount="indefinite" begin="0"></animate></rect></svg>`

`Fade Stagger Squares`
image

**Comment by vovakulikov** [comment]
Some updates 
- Instead of square animation I used skeleton blocks UI, it should help with visual jump a bit (as we discussed this in DM in slack with @taiyab)
- Animation over gradient in footer doc don't work in all browsers (because transition in gradient still isn't widely supported) I turned off transitions there (used gradient that I found in Figma but call it out that these are not standard one)

File: client/branded/src/search-ui/results/filters/components/dynamic-filter/SearchDynamicFilter.tsx [hunk 1]
```diff
@@ -131,7 +138,12 @@ export const SearchDynamicFilter: FC<SearchDynamicFilterProps> = ({
 
                 {filtersToShow.length === 0 && (
                     <small className={styles.description}>
-                        There are no {filterKind}s to show, try to use different search value
+                        <b>We couldn’t return a {filterKind} that matched your filter input.</b> Try a more expansive
+                        search{' '}
+                        <Button onClick={handleZeroStateButtonClick} className={styles.zeroStateSearchButton}>
+                            using the search bar
+                        </Button>{' '}
```
Comment (camdencheek) [inline client/branded/src/search-ui/results/filters/components/dynamic-filter/SearchDynamicFilter.tsx:145]: Could we add a button like this to the "Use search to see more" message when the filters are expanded? The copy there makes me think it's talking about the top-level search bar.

File: client/branded/src/search-ui/results/filters/NewSearchFilters.tsx [hunk 2]
```diff
@@ -98,6 +113,14 @@ export const NewSearchFilters: FC<NewSearchFiltersProps> = ({
         <div className={styles.scrollWrapper}>
             <FilterTypeList value={type} onSelect={handleFilterTypeChange} />
             <div className={styles.filters}>
+                {hadAnyFiltersButUtility && (
+                    <>
+                        <SearchFilterSkeleton />
+                        <SearchFilterSkeleton />
+                        <SearchFilterSkeleton />
```
Comment (camdencheek) [inline client/branded/src/search-ui/results/filters/NewSearchFilters.tsx:120]: Ooh, nice. These look great.

**Review by camdencheek — APPROVED** [review]
Whoops, thought I hit approve already

**Pull request merged**